File: 김채린/추가문제/프로그래머스/성격 유형 검사하기.java
import java.util.*;
import java.io.*;

class Solution {
    static char[][] type = new char[][]{{'R','T'}, {'C','F'},{'J','M'},{'A','N'}};
    public String solution(String[] survey, int[] choices) {
        HashMap<Character,Integer> map = new HashMap<>();
        for(int i=0; i<survey.length; i++){
            char[] chars = survey[i].toCharArray();
            char key;
            if (choices[i]>4) {
                key = chars[1];
            }else if (choices[i]<4){
                key=chars[0];
            }else {
                continue;
            }
            map.put(key, map.getOrDefault(key, 0) + Math.abs(choices[i]-4));
        }
        String result ="";
        for (char[] c: type){
            int a=map.getOrDefault(c[0],0);
            int b=map.getOrDefault(c[1],0);
            if (a<b){
                result+=c[1];
            }else {
                result+=c[0];
            }
        }
        return result;
    }
}

File: 정민선/프로그래머스/성격유형검사하기.py
def solution(survey, choices):
    answer = ''
    N = 4
    TEST = {
        'R':0, 'T':0,
        'C':0, 'F':0,
        'J':0, 'M':0,
        'A':0, 'N':0
    }
    for i in range(len(survey)):
        category = survey[i]
        result = choices[i]
        
        if result < 4:
            TEST[category[0]] += N-result
        elif result > 4:
            TEST[category[1]] += result-N
            
    answer += 'R' if TEST['R'] >= TEST['T'] else 'T'
    answer += 'C' if TEST['C'] >= TEST['F'] else 'F'
    answer += 'J' if TEST['J'] >= TEST['M'] else 'M'
    answer += 'A' if TEST['A'] >= TEST['N'] else 'N'
    return answer